internal/overlord/servstate/manager.go

```diff
@@ -146,6 +148,12 @@ func (m *ServiceManager) appendLayer(layer *plan.Layer) error {
 		return err
 	}
 	layer.Order = newOrder
+
+	// Remove the existing arguments of this layer's services
+	for _, service := range layer.Services {
+		delete(m.serviceArgs, service.Name)
+	}
```

Comment:
Yeah, this behaviour seems weird to me. I think we should just keep the service args when a layer is added. They're explicitly an override the user has requested, so seems weird to just drop them. Similar below.

Reply:
I am keeping this deletion, per our chat in Mattermost.